### ISSUE TITLE
VRT: A Video Restoration Transformer

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#138]
#### Sliding window-based methods
* Input multiple frames to generate a single HQ frame
* Process long video sequences in a sliding window fashion
* Each input frame is processed for multiple times in inference
→ inefficient feature utilization & increased computation cost

#### Recurrent methods
* Use previously reconstructed HQ frames for subsequent frame reconstruction
→ limited in parallelization for efficient distributed training and inference
→ bad at long-range temporal dependency modeling (one frame strongly affects next)
→ suffer from significant performance drop on few-frame videos

#### 본 논문에서 제안한 방법
* Parallel computation & long-range dependency modeling
  * Jointly extracts, aligns, fuses frame features at multiple scales
* Mutual attention for mutual alignment between frames
  * Soft version of image warping after implicit motion estimation
* Video super-resolution / Deblurring / Denoising

### VRT Overall framework
  ![image](https://user-images.githubusercontent.com/67627471/162620213-e628e6da-723c-475a-a0f6-93ad70be77c1.png)

### Feature Extraction
1. Extract shallow features 𝐼^𝑆𝐹 by a single spatial 2D convolution from the LQ sequence 𝐼^𝐿𝑄
2. Downsample the feature for S – 1 times (S: scales number) by squeezing each 2x2 neighborhood to the channel dimension
3. Reduce the channel number to the original number via a linear layer
4. Upsample the feature gradually by unsqueezing the feature back to its original size
→ Can extract features and deal with object by TMSA and parallel warping
5. Skip connections are added for features of same scales
6. Add several TMSA modules for further refinement and obtain the deep feature 𝐼^DF

#### Temporal Mutual Self Attention
* use mutual attention together with self attention
→ extract and preserve feature from the current frame

#### Parallel Warping
* Due to spatial window partitioning, the mutual attention mechanism may not be able to deal with large motions well
→ use feature warping at the end of each network stage to handle large motions

### Reconstruction
1. Reconstruct the HQ frames from the addition of shallow feature 𝐼^𝑆𝐹 and deep feature 𝐼^𝐷𝐹
2. Employ global residual learning and only predict the residual between the bilinearly upsampled LQ sequence and the ground-truth HQ sequence
→ ease the burden of feature learning
3. For video SR → sub-pixel conv layer to upsample the feature
  For video deblurring → single conv layer

#### Loss Function
* Charbonnier loss
  between the reconstructed HQ sequence 𝐼^𝑅𝐻𝑄 and the ground truth HQ sequence 𝐼^𝐻Q
  ![image](https://user-images.githubusercontent.com/67627471/162620449-7067d2e2-620e-4c1d-9639-80a832862927.png)

### Result
  ![image](https://user-images.githubusercontent.com/67627471/162620468-96dd3098-5cc1-459c-9942-f49b76f9ec03.png)

  ![image](https://user-images.githubusercontent.com/67627471/162620476-3ca6ae30-cc0d-4927-800e-fd5b4c619429.png)
